test/es6/destructuring_bugs.js

```diff
@@ -120,6 +120,129 @@ var tests = [
       assert.doesNotThrow(function () { eval("function foo(x = ({y = function (p) {}} = 'bar')) {}; foo();"); });
       assert.doesNotThrow(function () { eval("var foo = (x = ({y = function (p) {}} = 'bar')) => {}; foo();"); });
     }
+  },
+  {
+    name: "Destructuring empty patterns at param with arguments/eval at function body",
+    body: function () {
+        (function ({}, {}, {}, {}, {}, a) {
+            eval("");
+            assert.areEqual(arguments[1].x, 1);
+            assert.areEqual(a, 2);
+        })({}, {x:1}, {}, {}, {}, 2);
+        (function ({}, {}, {}, {}, {}, a) {
+            (function () {
+                eval("");
+            })();
+            assert.areEqual(arguments[1].x, 1);
+            assert.areEqual(a, 2);
+        })({}, {x:1}, {}, {}, {}, 2);
+        (function ({}, {}, {}, {}, {}, a) {
+            (function () {
+                a;
+            })();
+            eval("");
+            assert.areEqual(arguments[1].x, 1);
+            assert.areEqual(a, 2);
+        })({}, {x:1}, {}, {}, {}, 2);
+    }
+  },
+  {
+    name: "Destructuring patterns (multiple identifiers in the pattern) at param with arguments/eval at function body",
+    body: function () {
+        (function (x1, {x2, x3}, [x4, x5], x6) {
+            eval("");
+            assert.areEqual(arguments[2], [4, 5]);
+        })(1, {x2:2, x3:3}, [4, 5], 6);
+
+        (function (x1, {x2, x3}, [x4, x5], x6) {
+            var k = x1 + x2 + x3 + x4 + x5 + x6;
+            eval("");
+            assert.areEqual(arguments[2], [4, 5]);
+            assert.areEqual(k, 21);
+        })(1, {x2:2, x3:3}, [4, 5], 6);
+
+        (function (x1, {x2, x3}, [x4, x5], x6) {
+            (function() {
+              eval("");  
+            });
+            assert.areEqual(arguments[3], 6);
+            var k = x1 + x2 + x3 + x4 + x5 + x6;
+            assert.areEqual(k, 21);
+        })(1, {x2:2, x3:3}, [4, 5], 6);
+
+        (function (x1, {x2, x3}, [x4, x5], x6) {
+            (function() {
+                x3; x5; x6;
+            })();
+            var k = x1 + x2 + x3 + x4 + x5 + x6;
+            assert.areEqual(k, 21);
+        })(1, {x2:2, x3:3}, [4, 5], 6);
+
+        (function (x1, {x2, x3}, [x4, x5], x6) {
+            (function() {
+                x3; x5; x6;
+            })();
+            var k = x1 + x2 + x3 + x4 + x5 + x6;
+            assert.areEqual(arguments[3], 6);
+            assert.areEqual(k, 21);
+        })(1, {x2:2, x3:3}, [4, 5], 6);
+        
+        (function (x1, {x2, x3}, [x4, x5], x6) {
+            (function() {
+                assert.areEqual(x1 + x2 + x3 + x4 + x5 + x6, 21);
+            })();
+        })(1, {x2:2, x3:3}, [4, 5], 6);
+

```

Comment:
Can you add a test case with accessing arguments in the param scope? Also changing value of arguments in the param scope and validating it in the body scope?